build-scripts/MSVC.cmake

```
@@ -1,11 +1,53 @@
-if (NOT "$ENV{DevEnvDir}")
-    # Use Community Edition when not specified
-    set(ENV{DevEnvDir} "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\Common7\\IDE\\")
-endif()
-if(CMAKE_GENERATOR MATCHES ^Ninja)
-    # Ninja is provided by Microsoft but not in the Path
-    set(CMAKE_MAKE_PROGRAM $ENV{DevEnvDir}CommonExtensions\\Microsoft\\CMake\\Ninja\\ninja.exe CACHE PATH "")
-endif()
+#[=======================================================================[
+
+MSVC
+----
+
+Toolchain file for Microsoft Visual C++.
+
+Used by CMakePresets.json -> "toolchainFile".
+
+C++ flags used by all builds:
+
+/MP     cl.exe build with multiple processes
+/utf-8  set source and execution character sets to UTF-8
+/bigobj increase # of sections in object files
+/permissive-   to allow alternative operators ("and", "or", "not")
```

I see this is copy-pasted but I had to point out this is backwards :P

```suggestion
/permissive-   enforce more standards compliant behavior
```